src/main/java/baseball/domain/Balls.java

```diff
@@ -0,0 +1,59 @@
+package baseball.domain;
+
+import static java.util.stream.Collectors.collectingAndThen;
+import static java.util.stream.Collectors.toList;
+
+import java.util.Arrays;
+import java.util.List;
+import java.util.function.Predicate;
+
+public class Balls {
+
+    private static final int BALL_COUNT = 3;
+    private static final String DUPLICATES_EXCEPTION_MESSAGE = "[ERROR] 중복되지 않는 3자리의 숫자를 입력해주세요.";
+    private final List<Ball> balls;
+
+    private Balls(List<Ball> ballList) {
+        valdateDuplicates(ballList);
+        this.balls = ballList;
+    }
+
+    private void valdateDuplicates(List<Ball> ballList) {
+        if (hasDuplicatesInList(ballList)) {
+            throw new IllegalArgumentException(DUPLICATES_EXCEPTION_MESSAGE);
+        }
+    }
+
+    private boolean hasDuplicatesInList(List<Ball> ballList) {
+        int uniqueBallCount = (int) ballList.stream()
+            .distinct()
+            .count();
+        return uniqueBallCount < BALL_COUNT;
+    }
+
+    public GameResult getTryResultList(Balls playerBalls) {
+        return balls.stream()   // balls = answerBalls
+            .map(playerBalls::getTryResult)
+            .collect(collectingAndThen(toList(), GameResult::from));
+    }
```

Comment:
```suggestion
    public GameResult getTryResultList(Balls playerBalls) { // FIXME
        List<TryResult> result = balls.stream()   // balls = answerBalls
                .map(playerBalls::getTryResult)
                .toList();

        return GameResult.from(result);
    }

```

return 한 문장으로 메서드를 구현하는 것이 가독성 면에서는 떨어질 수도 있다고 생각해요.
리스트로 콜렉팅 하는 것과 팩터리 메서드를 호출하는 것이 한 라인에서 일어나니 동시에 두 가지일을 하는 것으로 느껴집니다.

콜렉팅해서, 객체를 만들어서 리턴하는 것이 이 메서드의 책임이라면, 각각을 개행함으로써 가독성을 더 높일 수도 있을 것 같네요 :D

Reply:
부끄럽지만 스트림의 사용에 취했던(?) 저의 과도한 스트림 사용이라고 느껴지네요😂😂
해로님이 제안한 코드가 확실히 가독성이 좋네요!